testing/test_config.py

```diff
@@ -2042,3 +2043,58 @@ def test_parse_warning_filter_failure(arg: str) -> None:
 
     with pytest.raises(warnings._OptionError):
         parse_warning_filter(arg, escape=True)
+
+
+def test_debug_file_name_is_suffixed_with_dot_log() -> None:
```

Comment:
Personally I would group those new tests into a class:

```python
class TestDebugOption:

    def test_suffixed_with_dot_log() -> None:
        ...
```

But that's just a matter of choice I guess.